src/osp/CoordinateSpaces/CartesianSimple.cpp

```diff
@@ -33,21 +33,47 @@ void CoordspaceCartesianSimple::update_exchange(
         Universe& rUni, CoordinateSpace& rSpace,
         CoordspaceCartesianSimple& rData)
 {
-    rData.reserve(rData.size() + rSpace.m_toAdd.size());
+    rData.reserve(rData.size() + rSpace.m_toAdd.size() - rSpace.m_toRemove.size());
 
-    auto view = rUni.get_reg().view<UCompCoordspaceIndex>();
+    auto coordIndexView = rUni.get_reg().view<UCompCoordspaceIndex>();
 
+    // Sort in descending order. If any of these are the last element, then
+    // then they may get invalidated in the following swaps.
+    sort(std::begin(rSpace.m_toRemove), std::end(rSpace.m_toRemove),
```

Comment:
Are you using ADL here?

Or some other implementation of sort?

Reply:
simply forgot an `std::`